openmdao/drivers/tests/test_differential_evolution_driver.py

```diff
@@ -667,6 +667,35 @@ def test_same_cons_and_obj(self):
         assert_near_equal(p.get_val('exec.z')[0], -900)
         assert_near_equal(p.get_val('exec.z')[50], -1000)
 
+    def test_inf_constraints(self):
+        from openmdao.core.constants import INF_BOUND
+
+        prob = om.Problem()
+        prob.model.add_subsystem('parab', Paraboloid(), promotes_inputs=['x', 'y'])
+
+        # define the component whose output will be constrained
+        prob.model.add_subsystem('const', om.ExecComp('g = x + y'), promotes_inputs=['x', 'y'])
+
+        # Design variables 'x' and 'y' span components, so we need to provide a common initial
+        # value for them.
+        prob.model.set_input_defaults('x', 3.0)
+        prob.model.set_input_defaults('y', -4.0)
+
+        # setup the optimization
+        prob.driver = om.DifferentialEvolutionDriver()
+
+        prob.model.add_design_var('x', lower=-INF_BOUND, upper=INF_BOUND)
+        prob.model.add_design_var('y', lower=-INF_BOUND, upper=INF_BOUND)
```

Comment:
Was there a user-case that motivated this? I'm not sure how you would encode a design variable with infinite bounds.

Reply:
it was reported by @hschilling so you could ask him...  I agree it seems like an unlikely scenario but it is a hole in the logic (un unhandled input).   I could imagine perhaps the bounds are set programmatically and INF_BOUND is the edge case...